[
  {
    "package": {
      "name": "cryptography",
      "version": "40.0.1",
      "ecosystem": "PyPI"
    },
    "dependency_groups": [
      "awscli-requirements"
    ],
    "vulnerabilities": [
      {
        "modified": "2024-02-23T05:16:42Z",
        "published": "2024-02-05T21:30:31Z",
        "schema_version": "1.6.0",
        "id": "GHSA-3ww4-gg4f-jr7f",
        "aliases": [
          "CVE-2023-50782"
        ],
        "summary": "Python Cryptography package vulnerable to Bleichenbacher timing oracle attack",
        "details": "A flaw was found in the python-cryptography package. This issue may allow a remote attacker to decrypt captured messages in TLS servers that use RSA key exchanges, which may lead to exposure of confidential or sensitive data.",
        "affected": [
          {
            "package": {
              "ecosystem": "PyPI",
              "name": "cryptography",
              "purl": "pkg:pypi/cryptography"
            },
            "ranges": [
              {
                "type": "ECOSYSTEM",
                "events": [
                  {
                    "introduced": "0"
                  },
                  {
                    "fixed": "42.0.0"
                  }
                ]
              }
            ],
            "versions": [
              "0.1",
              "0.2",
              "0.2.1",
              "0.2.2",
              "0.3",
              "0.4",
              "0.5",
              "0.5.1",
              "0.5.2",
              "0.5.3",
              "0.5.4",
              "0.6",
              "0.6.1",
              "0.7",
              "0.7.1",
              "0.7.2",
              "0.8",
              "0.8.1",
              "0.8.2",
              "0.9",
              "0.9.1",
              "0.9.2",
              "0.9.3",
              "1.0",
              "1.0.1",
              "1.0.2",
              "1.1",
              "1.1.1",
              "1.1.2",
              "1.2",
              "1.2.1",
              "1.2.2",
              "1.2.3",
              "1.3",
              "1.3.1",
              "1.3.2",
              "1.3.3",
              "1.3.4",
              "1.4",
              "1.5",
              "1.5.1",
              "1.5.2",
              "1.5.3",
              "1.6",
              "1.7",
              "1.7.1",
              "1.7.2",
              "1.8",
              "1.8.1",
              "1.8.2",
              "1.9",
              "2.0",
              "2.0.1",
              "2.0.2",
              "2.0.3",
              "2.1",
              "2.1.1",
              "2.1.2",
              "2.1.3",
              "2.1.4",
              "2.2",
              "2.2.1",
              "2.2.2",
              "2.3",
              "2.3.1",
              "2.4",
              "2.4.1",
              "2.4.2",
              "2.5",
              "2.6",
              "2.6.1",
              "2.7",
              "2.8",
              "2.9",
              "2.9.1",
              "2.9.2",
              "3.0",
              "3.1",
              "3.1.1",
              "3.2",
              "3.2.1",
              "3.3",
              "3.3.1",
              "3.3.2",
              "3.4",
              "3.4.1",
              "3.4.2",
              "3.4.3",
              "3.4.4",
              "3.4.5",
              "3.4.6",
              "3.4.7",
              "3.4.8",
              "35.0.0",
              "36.0.0",
              "36.0.1",
              "36.0.2",
              "37.0.0",
              "37.0.1",
              "37.0.2",
              "37.0.3",
              "37.0.4",
              "38.0.0",
              "38.0.1",
              "38.0.2",
              "38.0.3",
              "38.0.4",
              "39.0.0",
              "39.0.1",
              "39.0.2",
              "40.0.0",
              "40.0.1",
              "40.0.2",
              "41.0.0",
              "41.0.1",
              "41.0.2",
              "41.0.3",
              "41.0.4",
              "41.0.5",
              "41.0.6",
              "41.0.7"
            ],
            "database_specific": {
              "source": "https://github.com/github/advisory-database/blob/main/advisories/github-reviewed/2024/02/GHSA-3ww4-gg4f-jr7f/GHSA-3ww4-gg4f-jr7f.json"
            }
          }
        ],
        "severity": [
          {
            "type": "CVSS_V3",
            "score": "CVSS:3.1/AV:N/AC:L/PR:N/UI:N/S:U/C:H/I:N/A:N"
          }
        ],
        "references": [
          {
            "type": "ADVISORY",
            "url": "https://nvd.nist.gov/vuln/detail/CVE-2023-50782"
          },
          {
            "type": "WEB",
            "url": "https://github.com/pyca/cryptography/issues/9785"
          },
          {
            "type": "WEB",
            "url": "https://access.redhat.com/security/cve/CVE-2023-50782"
          },
          {
            "type": "WEB",
            "url": "https://bugzilla.redhat.com/show_bug.cgi?id=2254432"
          },
          {
            "type": "PACKAGE",
            "url": "https://github.com/pyca/cryptography"
          }
        ],
        "database_specific": {
          "cwe_ids": [
            "CWE-203",
            "CWE-208"
          ],
          "github_reviewed": true,
          "github_reviewed_at": "2024-02-05T23:04:50Z",
          "nvd_published_at": "2024-02-05T21:15:11Z",
          "severity": "HIGH"
        }
      },
      {
        "modified": "2024-02-21T23:01:06Z",
        "published": "2024-02-21T18:04:40Z",
        "schema_version": "1.6.0",
        "id": "GHSA-6vqw-3v5j-54x4",
        "aliases": [
          "CVE-2024-26130"
        ],
        "summary": "cryptography NULL pointer dereference with pkcs12.serialize_key_and_certificates when called with a non-matching certificate and private key and an hmac_hash override",
        "details": "If `pkcs12.serialize_key_and_certificates` is called with both:\n\n1. A certificate whose public key did not match the provided private key\n2. An `encryption_algorithm` with `hmac_hash` set (via `PrivateFormat.PKCS12.encryption_builder().hmac_hash(...)`\n\nThen a NULL pointer dereference would occur, crashing the Python process.\n\nThis has been resolved, and now a `ValueError` is properly raised.\n\nPatched in https://github.com/pyca/cryptography/pull/10423",
        "affected": [
          {
            "package": {
              "ecosystem": "PyPI",
              "name": "cryptography",
              "purl": "pkg:pypi/cryptography"
            },
            "ranges": [
              {
                "type": "ECOSYSTEM",
                "events": [
                  {
                    "introduced": "38.0.0"
                  },
                  {
                    "fixed": "42.0.4"
                  }
                ]
              }
            ],
            "versions": [
              "38.0.0",
              "38.0.1",
              "38.0.2",
              "38.0.3",
              "38.0.4",
              "39.0.0",
              "39.0.1",
              "39.0.2",
              "40.0.0",
              "40.0.1",
              "40.0.2",
              "41.0.0",
              "41.0.1",
              "41.0.2",
              "41.0.3",
              "41.0.4",
              "41.0.5",
              "41.0.6",
              "41.0.7",
              "42.0.0",
              "42.0.1",
              "42.0.2",
              "42.0.3"
            ],
            "database_specific": {
              "source": "https://github.com/github/advisory-database/blob/main/advisories/github-reviewed/2024/02/GHSA-6vqw-3v5j-54x4/GHSA-6vqw-3v5j-54x4.json"
            }
          }
        ],
        "severity": [
          {
            "type": "CVSS_V3",
            "score": "CVSS:3.1/AV:N/AC:L/PR:N/UI:N/S:U/C:N/I:N/A:H"
          }
        ],
        "references": [
          {
            "type": "WEB",
            "url": "https://github.com/pyca/cryptography/security/advisories/GHSA-6vqw-3v5j-54x4"
          },
          {
            "type": "ADVISORY",
            "url": "https://nvd.nist.gov/vuln/detail/CVE-2024-26130"
          },
          {
            "type": "WEB",
            "url": "https://github.com/pyca/cryptography/pull/10423"
          },
          {
            "type": "WEB",
            "url": "https://github.com/pyca/cryptography/commit/97d231672763cdb5959a3b191e692a362f1b9e55"
          },
          {
            "type": "PACKAGE",
            "url": "https://github.com/pyca/cryptography"
          }
        ],
        "database_specific": {
          "cwe_ids": [
            "CWE-476"
          ],
          "github_reviewed": true,
          "github_reviewed_at": "2024-02-21T18:04:40Z",
          "nvd_published_at": "2024-02-21T17:15:09Z",
          "severity": "HIGH"
        }
      },
      {
        "modified": "2024-05-02T13:32:49Z",
        "published": "2024-01-26T09:30:23Z",
        "schema_version": "1.6.0",
        "id": "GHSA-9v9h-cgj8-h64p",
        "aliases": [
          "CVE-2024-0727"
        ],
        "summary": "Null pointer dereference in PKCS12 parsing",
        "details": "Issue summary: Processing a maliciously formatted PKCS12 file may lead OpenSSL\nto crash leading to a potential Denial of Service attack\n\nImpact summary: Applications loading files in the PKCS12 format from untrusted\nsources might terminate abruptly.\n\nA file in PKCS12 format can contain certificates and keys and may come from an\nuntrusted source. The PKCS12 specification allows certain fields to be NULL, but\nOpenSSL does not correctly check for this case. This can lead to a NULL pointer\ndereference that results in OpenSSL crashing. If an application processes PKCS12\nfiles from an untrusted source using the OpenSSL APIs then that application will\nbe vulnerable to this issue.\n\nOpenSSL APIs that are vulnerable to this are: PKCS12_parse(),\nPKCS12_unpack_p7data(), PKCS12_unpack_p7encdata(), PKCS12_unpack_authsafes()\nand PKCS12_newpass().\n\nWe have also fixed a similar issue in SMIME_write_PKCS7(). However since this\nfunction is related to writing data we do not consider it security significant.\n\nThe FIPS modules in 3.2, 3.1 and 3.0 are not affected by this issue.",
        "affected": [
          {
            "package": {
              "ecosystem": "PyPI",
              "name": "cryptography",
              "purl": "pkg:pypi/cryptography"
            },
            "ranges": [
              {
                "type": "ECOSYSTEM",
                "events": [
                  {
                    "introduced": "0"
                  },
                  {
                    "fixed": "42.0.2"
                  }
                ]
              }
            ],
            "versions": [
              "0.1",
              "0.2",
              "0.2.1",
              "0.2.2",
              "0.3",
              "0.4",
              "0.5",
              "0.5.1",
              "0.5.2",
              "0.5.3",
              "0.5.4",
              "0.6",
              "0.6.1",
              "0.7",
              "0.7.1",
              "0.7.2",
              "0.8",
              "0.8.1",
              "0.8.2",
              "0.9",
              "0.9.1",
              "0.9.2",
              "0.9.3",
              "1.0",
              "1.0.1",
              "1.0.2",
              "1.1",
              "1.1.1",
              "1.1.2",
              "1.2",
              "1.2.1",
              "1.2.2",
              "1.2.3",
              "1.3",
              "1.3.1",
              "1.3.2",
              "1.3.3",
              "1.3.4",
              "1.4",
              "1.5",
              "1.5.1",
              "1.5.2",
              "1.5.3",
              "1.6",
              "1.7",
              "1.7.1",
              "1.7.2",
              "1.8",
              "1.8.1",
              "1.8.2",
              "1.9",
              "2.0",
              "2.0.1",
              "2.0.2",
              "2.0.3",
              "2.1",
              "2.1.1",
              "2.1.2",
              "2.1.3",
              "2.1.4",
              "2.2",
              "2.2.1",
              "2.2.2",
              "2.3",
              "2.3.1",
              "2.4",
              "2.4.1",
              "2.4.2",
              "2.5",
              "2.6",
              "2.6.1",
              "2.7",
              "2.8",
              "2.9",
              "2.9.1",
              "2.9.2",
              "3.0",
              "3.1",
              "3.1.1",
              "3.2",
              "3.2.1",
              "3.3",
              "3.3.1",
              "3.3.2",
              "3.4",
              "3.4.1",
              "3.4.2",
              "3.4.3",
              "3.4.4",
              "3.4.5",
              "3.4.6",
              "3.4.7",
              "3.4.8",
              "35.0.0",
              "36.0.0",
              "36.0.1",
              "36.0.2",
              "37.0.0",
              "37.0.1",
              "37.0.2",
              "37.0.3",
              "37.0.4",
              "38.0.0",
              "38.0.1",
              "38.0.2",
              "38.0.3",
              "38.0.4",
              "39.0.0",
              "39.0.1",
              "39.0.2",
              "40.0.0",
              "40.0.1",
              "40.0.2",
              "41.0.0",
              "41.0.1",
              "41.0.2",
              "41.0.3",
              "41.0.4",
              "41.0.5",
              "41.0.6",
              "41.0.7",
              "42.0.0",
              "42.0.1"
            ],
            "database_specific": {
              "source": "https://github.com/github/advisory-database/blob/main/advisories/github-reviewed/2024/01/GHSA-9v9h-cgj8-h64p/GHSA-9v9h-cgj8-h64p.json"
            },
            "ecosystem_specific": {
              "affected_functions": [
                "cryptography.hazmat.backends.openssl.backend.load_pkcs12"
              ]
            }
          }
        ],
        "severity": [
          {
            "type": "CVSS_V3",
            "score": "CVSS:3.1/AV:L/AC:L/PR:N/UI:R/S:U/C:N/I:N/A:H"
          }
        ],
        "references": [
          {
            "type": "ADVISORY",
            "url": "https://nvd.nist.gov/vuln/detail/CVE-2024-0727"
          },
          {
            "type": "WEB",
            "url": "https://github.com/github/advisory-database/pull/3472"
          },
          {
            "type": "WEB",
            "url": "https://github.com/openssl/openssl/pull/23362"
          },
          {
            "type": "WEB",
            "url": "https://github.com/alexcrichton/openssl-src-rs/commit/add20f73b6b42be7451af2e1044d4e0e778992b2"
          },
          {
            "type": "WEB",
            "url": "https://github.com/openssl/openssl/commit/09df4395b5071217b76dc7d3d2e630eb8c5a79c2"
          },
          {
            "type": "WEB",
            "url": "https://github.com/openssl/openssl/commit/775acfdbd0c6af9ac855f34969cdab0c0c90844a"
          },
          {
            "type": "WEB",
            "url": "https://github.com/openssl/openssl/commit/d135eeab8a5dbf72b3da5240bab9ddb7678dbd2c"
          },
          {
            "type": "WEB",
            "url": "https://github.com/pyca/cryptography/commit/3519591d255d4506fbcd0d04037d45271903c64d"
          },
          {
            "type": "WEB",
            "url": "https://github.openssl.org/openssl/extended-releases/commit/03b3941d60c4bce58fab69a0c22377ab439bc0e8"
          },
          {
            "type": "WEB",
            "url": "https://github.openssl.org/openssl/extended-releases/commit/aebaa5883e31122b404e450732dc833dc9dee539"
          },
          {
            "type": "WEB",
            "url": "https://security.netapp.com/advisory/ntap-20240208-0006"
          },
          {
            "type": "WEB",
            "url": "https://www.openssl.org/news/secadv/20240125.txt"
          },
          {
            "type": "WEB",
            "url": "http://www.openwall.com/lists/oss-security/2024/03/11/1"
          }
        ],
        "database_specific": {
          "cwe_ids": [],
          "github_reviewed": true,
          "github_reviewed_at": "2024-02-16T20:48:36Z",
          "nvd_published_at": "2024-01-26T09:15:07Z",
          "severity": "MODERATE"
        }
      },
      {
        "modified": "2024-02-16T08:08:09Z",
        "published": "2023-07-14T21:31:08Z",
        "schema_version": "1.6.0",
        "id": "GHSA-cf7p-gm2m-833m",
        "aliases": [
          "CVE-2023-38325",
          "PYSEC-2023-112"
        ],
        "summary": "cryptography mishandles SSH certificates",
        "details": "The cryptography package before 41.0.2 for Python mishandles SSH certificates that have critical options.",
        "affected": [
          {
            "package": {
              "ecosystem": "PyPI",
              "name": "cryptography",
              "purl": "pkg:pypi/cryptography"
            },
            "ranges": [
              {
                "type": "ECOSYSTEM",
                "events": [
                  {
                    "introduced": "40.0.0"
                  },
                  {
                    "fixed": "41.0.2"
                  }
                ]
              }
            ],
            "versions": [
              "40.0.0",
              "40.0.1",
              "40.0.2",
              "41.0.0",
              "41.0.1"
            ],
            "database_specific": {
              "source": "https://github.com/github/advisory-database/blob/main/advisories/github-reviewed/2023/07/GHSA-cf7p-gm2m-833m/GHSA-cf7p-gm2m-833m.json"
            }
          }
        ],
        "severity": [
          {
            "type": "CVSS_V3",
            "score": "CVSS:3.1/AV:N/AC:L/PR:N/UI:N/S:U/C:N/I:H/A:N"
          }
        ],
        "references": [
          {
            "type": "ADVISORY",
            "url": "https://nvd.nist.gov/vuln/detail/CVE-2023-38325"
          },
          {
            "type": "WEB",
            "url": "https://github.com/pyca/cryptography/issues/9207"
          },
          {
            "type": "WEB",
            "url": "https://github.com/pyca/cryptography/pull/7960"
          },
          {
            "type": "WEB",
            "url": "https://github.com/pyca/cryptography/pull/9208"
          },
          {
            "type": "WEB",
            "url": "https://github.com/pyca/cryptography/commit/1ca7adc97b76a9dfbd3d850628b613eb93b78fc3"
          },
          {
            "type": "PACKAGE",
            "url": "https://github.com/pyca/cryptography"
          },
          {
            "type": "WEB",
            "url": "https://github.com/pyca/cryptography/compare/41.0.1...41.0.2"
          },
          {
            "type": "WEB",
            "url": "https://github.com/pypa/advisory-database/tree/main/vulns/cryptography/PYSEC-2023-112.yaml"
          },
          {
            "type": "WEB",
            "url": "https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/NMCCTYY3CSNQBFFYYC5DAV6KATHWCUZK"
          },
          {
            "type": "WEB",
            "url": "https://pypi.org/project/cryptography/#history"
          },
          {
            "type": "WEB",
            "url": "https://security.netapp.com/advisory/ntap-20230824-0010"
          }
        ],
        "database_specific": {
          "cwe_ids": [
            "CWE-295"
          ],
          "github_reviewed": true,
          "github_reviewed_at": "2023-07-14T22:14:26Z",
          "nvd_published_at": "2023-07-14T20:15:09Z",
          "severity": "HIGH"
        }
      },
      {
        "modified": "2024-02-20T18:17:05Z",
        "published": "2023-11-28T20:46:46Z",
        "schema_version": "1.6.0",
        "id": "GHSA-jfhm-5ghh-2f97",
        "aliases": [
          "CVE-2023-49083",
          "PYSEC-2023-254"
        ],
        "summary": "cryptography vulnerable to NULL-dereference when loading PKCS7 certificates",
        "details": "### Summary\n\nCalling `load_pem_pkcs7_certificates` or `load_der_pkcs7_certificates` could lead to a NULL-pointer dereference and segfault.\n\n### PoC\nHere is a Python code that triggers the issue:\n```python\nfrom cryptography.hazmat.primitives.serialization.pkcs7 import load_der_pkcs7_certificates, load_pem_pkcs7_certificates\n\npem_p7 = b\"\"\"\n-----BEGIN PKCS7-----\nMAsGCSqGSIb3DQEHAg==\n-----END PKCS7-----\n\"\"\"\n\nder_p7 = b\"\\x30\\x0B\\x06\\x09\\x2A\\x86\\x48\\x86\\xF7\\x0D\\x01\\x07\\x02\"\n\nload_pem_pkcs7_certificates(pem_p7)\nload_der_pkcs7_certificates(der_p7)\n```\n\n### Impact\nExploitation of this vulnerability poses a serious risk of Denial of Service (DoS) for any application attempting to deserialize a PKCS7 blob/certificate. The consequences extend to potential disruptions in system availability and stability.",
        "affected": [
          {
            "package": {
              "ecosystem": "PyPI",
              "name": "cryptography",
              "purl": "pkg:pypi/cryptography"
            },
            "ranges": [
              {
                "type": "ECOSYSTEM",
                "events": [
                  {
                    "introduced": "3.1"
                  },
                  {
                    "fixed": "41.0.6"
                  }
                ]
              }
            ],
            "versions": [
              "3.1",
              "3.1.1",
              "3.2",
              "3.2.1",
              "3.3",
              "3.3.1",
              "3.3.2",
              "3.4",
              "3.4.1",
              "3.4.2",
              "3.4.3",
              "3.4.4",
              "3.4.5",
              "3.4.6",
              "3.4.7",
              "3.4.8",
              "35.0.0",
              "36.0.0",
              "36.0.1",
              "36.0.2",
              "37.0.0",
              "37.0.1",
              "37.0.2",
              "37.0.3",
              "37.0.4",
              "38.0.0",
              "38.0.1",
              "38.0.2",
              "38.0.3",
              "38.0.4",
              "39.0.0",
              "39.0.1",
              "39.0.2",
              "40.0.0",
              "40.0.1",
              "40.0.2",
              "41.0.0",
              "41.0.1",
              "41.0.2",
              "41.0.3",
              "41.0.4",
              "41.0.5"
            ],
            "database_specific": {
              "source": "https://github.com/github/advisory-database/blob/main/advisories/github-reviewed/2023/11/GHSA-jfhm-5ghh-2f97/GHSA-jfhm-5ghh-2f97.json"
            },
            "ecosystem_specific": {
              "affected_functions": [
                "cryptography.hazmat.primitives.serialization.pkcs7.load_der_pkcs7_certificates",
                "cryptography.hazmat.primitives.serialization.pkcs7.load_pem_pkcs7_certificates"
              ]
            }
          }
        ],
        "severity": [
          {
            "type": "CVSS_V3",
            "score": "CVSS:3.1/AV:N/AC:H/PR:N/UI:N/S:U/C:N/I:N/A:H"
          }
        ],
        "references": [
          {
            "type": "WEB",
            "url": "https://github.com/pyca/cryptography/security/advisories/GHSA-jfhm-5ghh-2f97"
          },
          {
            "type": "ADVISORY",
            "url": "https://nvd.nist.gov/vuln/detail/CVE-2023-49083"
          },
          {
            "type": "WEB",
            "url": "https://github.com/pyca/cryptography/pull/9926"
          },
          {
            "type": "WEB",
            "url": "https://github.com/pyca/cryptography/commit/f09c261ca10a31fe41b1262306db7f8f1da0e48a"
          },
          {
            "type": "PACKAGE",
            "url": "https://github.com/pyca/cryptography"
          },
          {
            "type": "WEB",
            "url": "https://github.com/pypa/advisory-database/tree/main/vulns/cryptography/PYSEC-2023-254.yaml"
          },
          {
            "type": "WEB",
            "url": "https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/QMNTYMUGFJSDBYBU22FUYBHFRZODRKXV"
          },
          {
            "type": "WEB",
            "url": "http://www.openwall.com/lists/oss-security/2023/11/29/2"
          }
        ],
        "database_specific": {
          "cwe_ids": [
            "CWE-476"
          ],
          "github_reviewed": true,
          "github_reviewed_at": "2023-11-28T20:46:46Z",
          "nvd_published_at": "2023-11-29T19:15:07Z",
          "severity": "MODERATE"
        }
      },
      {
        "modified": "2023-11-08T04:13:07Z",
        "published": "2023-07-14T20:15:00Z",
        "schema_version": "1.6.0",
        "id": "PYSEC-2023-112",
        "aliases": [
          "CVE-2023-38325",
          "GHSA-cf7p-gm2m-833m"
        ],
        "details": "The cryptography package before 41.0.2 for Python mishandles SSH certificates that have critical options.",
        "affected": [
          {
            "package": {
              "ecosystem": "PyPI",
              "name": "cryptography",
              "purl": "pkg:pypi/cryptography"
            },
            "ranges": [
              {
                "type": "ECOSYSTEM",
                "events": [
                  {
                    "introduced": "40.0.0"
                  },
                  {
                    "fixed": "41.0.2"
                  }
                ]
              }
            ],
            "versions": [
              "40.0.0",
              "40.0.1",
              "40.0.2",
              "41.0.0",
              "41.0.1"
            ],
            "database_specific": {
              "source": "https://github.com/pypa/advisory-database/blob/main/vulns/cryptography/PYSEC-2023-112.yaml"
            }
          }
        ],
        "references": [
          {
            "type": "PACKAGE",
            "url": "https://pypi.org/project/cryptography/#history"
          },
          {
            "type": "WEB",
            "url": "https://github.com/pyca/cryptography/compare/41.0.1...41.0.2"
          },
          {
            "type": "REPORT",
            "url": "https://github.com/pyca/cryptography/issues/9207"
          },
          {
            "type": "WEB",
            "url": "https://github.com/pyca/cryptography/pull/9208"
          }
        ]
      },
      {
        "modified": "2024-02-17T07:41:40Z",
        "published": "2023-11-29T19:15:00Z",
        "schema_version": "1.6.0",
        "id": "PYSEC-2023-254",
        "aliases": [
          "CVE-2023-49083",
          "GHSA-jfhm-5ghh-2f97"
        ],
        "details": "cryptography is a package designed to expose cryptographic primitives and recipes to Python developers. Calling `load_pem_pkcs7_certificates` or `load_der_pkcs7_certificates` could lead to a NULL-pointer dereference and segfault. Exploitation of this vulnerability poses a serious risk of Denial of Service (DoS) for any application attempting to deserialize a PKCS7 blob/certificate. The consequences extend to potential disruptions in system availability and stability. This vulnerability has been patched in version 41.0.6.",
        "affected": [
          {
            "package": {
              "ecosystem": "PyPI",
              "name": "cryptography",
              "purl": "pkg:pypi/cryptography"
            },
            "ranges": [
              {
                "type": "GIT",
                "events": [
                  {
                    "introduced": "0"
                  },
                  {
                    "fixed": "f09c261ca10a31fe41b1262306db7f8f1da0e48a"
                  }
                ],
                "repo": "https://github.com/pyca/cryptography"
              },
              {
                "type": "ECOSYSTEM",
                "events": [
                  {
                    "introduced": "3.1"
                  },
                  {
                    "fixed": "41.0.6"
                  }
                ]
              }
            ],
            "versions": [
              "3.1",
              "3.1.1",
              "3.2",
              "3.2.1",
              "3.3",
              "3.3.1",
              "3.3.2",
              "3.4",
              "3.4.1",
              "3.4.2",
              "3.4.3",
              "3.4.4",
              "3.4.5",
              "3.4.6",
              "3.4.7",
              "3.4.8",
              "35.0.0",
              "36.0.0",
              "36.0.1",
              "36.0.2",
              "37.0.0",
              "37.0.1",
              "37.0.2",
              "37.0.3",
              "37.0.4",
              "38.0.0",
              "38.0.1",
              "38.0.2",
              "38.0.3",
              "38.0.4",
              "39.0.0",
              "39.0.1",
              "39.0.2",
              "40.0.0",
              "40.0.1",
              "40.0.2",
              "41.0.0",
              "41.0.1",
              "41.0.2",
              "41.0.3",
              "41.0.4",
              "41.0.5"
            ],
            "database_specific": {
              "source": "https://github.com/pypa/advisory-database/blob/main/vulns/cryptography/PYSEC-2023-254.yaml"
            }
          }
        ],
        "severity": [
          {
            "type": "CVSS_V3",
            "score": "CVSS:3.1/AV:N/AC:L/PR:N/UI:N/S:U/C:N/I:N/A:H"
          }
        ],
        "references": [
          {
            "type": "ADVISORY",
            "url": "https://github.com/pyca/cryptography/security/advisories/GHSA-jfhm-5ghh-2f97"
          },
          {
            "type": "FIX",
            "url": "https://github.com/pyca/cryptography/pull/9926"
          },
          {
            "type": "FIX",
            "url": "https://github.com/pyca/cryptography/commit/f09c261ca10a31fe41b1262306db7f8f1da0e48a"
          },
          {
            "type": "WEB",
            "url": "https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/QMNTYMUGFJSDBYBU22FUYBHFRZODRKXV/"
          }
        ]
      }
    ],
    "groups": [
      {
        "ids": [
          "GHSA-3ww4-gg4f-jr7f"
        ],
        "aliases": [
          "CVE-2023-50782",
          "GHSA-3ww4-gg4f-jr7f"
        ],
        "max_severity": "7.5"
      },
      {
        "ids": [
          "GHSA-6vqw-3v5j-54x4"
        ],
        "aliases": [
          "CVE-2024-26130",
          "GHSA-6vqw-3v5j-54x4"
        ],
        "max_severity": "7.5"
      },
      {
        "ids": [
          "GHSA-9v9h-cgj8-h64p"
        ],
        "aliases": [
          "CVE-2024-0727",
          "GHSA-9v9h-cgj8-h64p"
        ],
        "max_severity": "5.5"
      },
      {
        "ids": [
          "GHSA-cf7p-gm2m-833m",
          "PYSEC-2023-112"
        ],
        "aliases": [
          "CVE-2023-38325",
          "GHSA-cf7p-gm2m-833m",
          "PYSEC-2023-112"
        ],
        "max_severity": "7.5"
      },
      {
        "ids": [
          "GHSA-jfhm-5ghh-2f97",
          "PYSEC-2023-254"
        ],
        "aliases": [
          "CVE-2023-49083",
          "GHSA-jfhm-5ghh-2f97",
          "PYSEC-2023-254"
        ],
        "max_severity": "7.5"
      }
    ]
  },
  {
    "package": {
      "name": "urllib3",
      "version": "1.26.18",
      "ecosystem": "PyPI"
    },
    "dependency_groups": [
      "awscli-requirements"
    ],
    "vulnerabilities": [
      {
        "modified": "2024-06-17T22:12:25Z",
        "published": "2024-06-17T21:37:20Z",
        "schema_version": "1.6.0",
        "id": "GHSA-34jh-p97f-mpxf",
        "aliases": [
          "CVE-2024-37891"
        ],
        "summary": "urllib3's Proxy-Authorization request header isn't stripped during cross-origin redirects ",
        "details": "When using urllib3's proxy support with `ProxyManager`, the `Proxy-Authorization` header is only sent to the configured proxy, as expected.\n\nHowever, when sending HTTP requests *without* using urllib3's proxy support, it's possible to accidentally configure the `Proxy-Authorization` header even though it won't have any effect as the request is not using a forwarding proxy or a tunneling proxy. In those cases, urllib3 doesn't treat the `Proxy-Authorization` HTTP header as one carrying authentication material and thus doesn't strip the header on cross-origin redirects.\n\nBecause this is a highly unlikely scenario, we believe the severity of this vulnerability is low for almost all users. Out of an abundance of caution urllib3 will automatically strip the `Proxy-Authorization` header during cross-origin redirects to avoid the small chance that users are doing this on accident.\n\nUsers should use urllib3's proxy support or disable automatic redirects to achieve safe processing of the `Proxy-Authorization` header, but we still decided to strip the header by default in order to further protect users who aren't using the correct approach.\n\n## Affected usages\n\nWe believe the number of usages affected by this advisory is low. It requires all of the following to be true to be exploited:\n\n* Setting the `Proxy-Authorization` header without using urllib3's built-in proxy support.\n* Not disabling HTTP redirects.\n* Either not using an HTTPS origin server or for the proxy or target origin to redirect to a malicious origin.\n\n## Remediation\n\n* Using the `Proxy-Authorization` header with urllib3's `ProxyManager`.\n* Disabling HTTP redirects using `redirects=False` when sending requests.\n* Not using the `Proxy-Authorization` header.",
        "affected": [
          {
            "package": {
              "ecosystem": "PyPI",
              "name": "urllib3",
              "purl": "pkg:pypi/urllib3"
            },
            "ranges": [
              {
                "type": "ECOSYSTEM",
                "events": [
                  {
                    "introduced": "0"
                  },
                  {
                    "fixed": "1.26.19"
                  }
                ]
              }
            ],
            "versions": [
              "0.2",
              "0.3",
              "0.3.1",
              "0.4.0",
              "0.4.1",
              "1.0",
              "1.0.1",
              "1.0.2",
              "1.1",
              "1.10",
              "1.10.1",
              "1.10.2",
              "1.10.3",
              "1.10.4",
              "1.11",
              "1.12",
              "1.13",
              "1.13.1",
              "1.14",
              "1.15",
              "1.15.1",
              "1.16",
              "1.17",
              "1.18",
              "1.18.1",
              "1.19",
              "1.19.1",
              "1.2",
              "1.2.1",
              "1.2.2",
              "1.20",
              "1.21",
              "1.21.1",
              "1.22",
              "1.23",
              "1.24",
              "1.24.1",
              "1.24.2",
              "1.24.3",
              "1.25",
              "1.25.1",
              "1.25.10",
              "1.25.11",
              "1.25.2",
              "1.25.3",
              "1.25.4",
              "1.25.5",
              "1.25.6",
              "1.25.7",
              "1.25.8",
              "1.25.9",
              "1.26.0",
              "1.26.1",
              "1.26.10",
              "1.26.11",
              "1.26.12",
              "1.26.13",
              "1.26.14",
              "1.26.15",
              "1.26.16",
              "1.26.17",
              "1.26.18",
              "1.26.2",
              "1.26.3",
              "1.26.4",
              "1.26.5",
              "1.26.6",
              "1.26.7",
              "1.26.8",
              "1.26.9",
              "1.3",
              "1.4",
              "1.5",
              "1.6",
              "1.7",
              "1.7.1",
              "1.8",
              "1.8.2",
              "1.8.3",
              "1.9",
              "1.9.1"
            ],
            "database_specific": {
              "source": "https://github.com/github/advisory-database/blob/main/advisories/github-reviewed/2024/06/GHSA-34jh-p97f-mpxf/GHSA-34jh-p97f-mpxf.json"
            }
          },
          {
            "package": {
              "ecosystem": "PyPI",
              "name": "urllib3",
              "purl": "pkg:pypi/urllib3"
            },
            "ranges": [
              {
                "type": "ECOSYSTEM",
                "events": [
                  {
                    "introduced": "2.0.0"
                  },
                  {
                    "fixed": "2.2.2"
                  }
                ]
              }
            ],
            "versions": [
              "2.0.0",
              "2.0.1",
              "2.0.2",
              "2.0.3",
              "2.0.4",
              "2.0.5",
              "2.0.6",
              "2.0.7",
              "2.1.0",
              "2.2.0",
              "2.2.1"
            ],
            "database_specific": {
              "source": "https://github.com/github/advisory-database/blob/main/advisories/github-reviewed/2024/06/GHSA-34jh-p97f-mpxf/GHSA-34jh-p97f-mpxf.json"
            }
          }
        ],
        "severity": [
          {
            "type": "CVSS_V3",
            "score": "CVSS:3.1/AV:N/AC:H/PR:H/UI:N/S:U/C:H/I:N/A:N"
          }
        ],
        "references": [
          {
            "type": "WEB",
            "url": "https://github.com/urllib3/urllib3/security/advisories/GHSA-34jh-p97f-mpxf"
          },
          {
            "type": "ADVISORY",
            "url": "https://nvd.nist.gov/vuln/detail/CVE-2024-37891"
          },
          {
            "type": "WEB",
            "url": "https://github.com/urllib3/urllib3/commit/40b6d1605814dd1db0a46e202d6e56f2e4c9a468"
          },
          {
            "type": "WEB",
            "url": "https://github.com/urllib3/urllib3/commit/accff72ecc2f6cf5a76d9570198a93ac7c90270e"
          },
          {
            "type": "PACKAGE",
            "url": "https://github.com/urllib3/urllib3"
          }
        ],
        "database_specific": {
          "cwe_ids": [
            "CWE-669"
          ],
          "github_reviewed": true,
          "github_reviewed_at": "2024-06-17T21:37:20Z",
          "nvd_published_at": "2024-06-17T20:15:13Z",
          "severity": "MODERATE"
        }
      }
    ],
    "groups": [
      {
        "ids": [
          "GHSA-34jh-p97f-mpxf"
        ],
        "aliases": [
          "CVE-2024-37891",
          "GHSA-34jh-p97f-mpxf"
        ],
        "max_severity": "4.4"
      }
    ]
  }
]